src/dc_imex.rs

```diff
@@ -514,7 +514,7 @@ pub unsafe fn dc_normalize_setup_code(
 
 #[allow(non_snake_case)]
 pub unsafe fn dc_job_do_DC_JOB_IMEX_IMAP(context: &Context, job: *mut dc_job_t) {
-    let mut current_block: u64;
+    let mut key_exists = true;
```

Comment:
please init to false

Reply:
We didn't init to false in moster other goto refactors, initing to false forces us to keep a lot of else branches which do nothing besides setting this to true.

Reply:
but that is the whole point, to keep accurate track of if this should be set to to true or not :/

Reply:
I guess this comes down to the variable name, if it is a `ok_to_continue` it is different, than saying `key_exists`.

Reply:
I don't care too much, the goto methods are still quite unreadable because of the right shift and should get refactored quite soon.